jerry-core/parser/js/js-scanner-util.c

```diff
@@ -1752,6 +1752,37 @@ scanner_is_global_context_needed (parser_context_t *context_p) /**< context */
   return false;
 } /* scanner_is_global_context_needed */
 
+/**
+ * Try to scan/parse the ".target" part in the "new.target" expression.
+ *
+ * Upon exiting with "true" the current token will point to the "target"
+ * literal.
+ *
+ * If the "target" literal is not after the "new." then a scanner/parser
+ * error will be raised.
+ *
+ * @returns true if the ".target" part was found
+ *          false if there is no "." after the new.
+ */
+bool
+scanner_try_scan_new_target (parser_context_t *context_p) /**< parser/scanner context */
+{
+  JERRY_ASSERT (context_p->token.type == LEXER_KEYW_NEW);
+
+  if (lexer_check_next_character (context_p, LIT_CHAR_DOT))
+  {
+    lexer_consume_next_character (context_p);
```

Comment:
I think this might affect the column info in the error. So I would call `lexer_next_token` here and check that the dot is a single dot character. This way we could get a correct column when three dots are used. I think this is a rare case, so the overall perf overhead is small.

Reply:
Thanks, updated!